src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+Our `then` method is simply a "glorified callback aggregator".
+
+The implementation skeleton for Infusion's promises was taken from a code sample by John Hann (unscriptable) in this [gist](https://gist.github.com/unscriptable/814052) - full credit and thanks,
+and please read the gist for some further commentary and coverage of limitations.
+
+
+## Core Promises Utility API
+
+The library implements a few utilities without which it is inconvenient to use promises:
+
+### fluid.isPromise(totest)
+
+* `totest {Any}` An object to be checked for being a promise
+* Returns: `{Boolean}` If `totest` has a member `then` of type `Function`, returns `true`.
+
+Determines whether an object is a promise, for our purposes. Any object with a member `then` of type `Function` passes this test.
+This includes essentially every known variety, including jQuery promises. This test in fact identifies what in other libraries is
+termed a "foreign thenable".
+
+### fluid.toPromise(value)
+
+* `value {Any}` A value to be converted ("hoisted") to a promise
+* Returns: `{Promise}` If the supplied value is already a promise, it is returned unchanged. Otherwise a fresh promise is created with the value as resolution and returned.
+
+Coerces any value to a promise. If it is already a promise, it is returned unchanged. Otherwise, `toPromise` constructs a fresh promise resolving to the supplied value.
+
+
+### fluid.promise.follow(source, target)
+
+* `source {Promise}` A promise which is to be followed in its resolution.
+* `target {Promise}` A promise which will follow the `source` in its resolution. `target` will receive a call to `then` causing it to resolve when `source` resolves,
+and reject when `source` rejects. 
+
+Chains the resolution methods of one promise (target) so that they follow those of another (source).
+That is, whenever source resolves, target will resolve, or when source rejects, target will reject, with the
+same payloads in each case.
+
+### fluid.promise.map(source, func)
+
+Returns a promise whose resolved value is mapped from the source promise or value by the supplied function. If the input value is not a promise, it will
+be converted first to a promise via `fluid.toPromise()`. If the input promise rejects, its rejection reason will be propagated unmapped. 
+
+* `source {Object|Promise}` An object or promise whose value is to be mapped (if an object, will be converted first to a promise via `fluid.toPromise()`)
+* `func {Function: ({Any})  →  Any}` A function which will map the resolved promise value
+* Returns: `{Promise}` A promise for the resolved mapped value.
+ 
+## Promise algorithms
+
+The only currently implemented promise algorithms are based around a core skeleton operating an array of promises in a linear sequence. These 
+are responsive to an additional element of our promises API, the  [`fluid.accumulateRejectionReason`](#promise.accumulaterejectionreason-error-) "inverse API" described below.
```

Comment:
This link text (fluid.accumulateRejectionReason) is incorrect, and the link itself is broken